src/go/rucio-dataset-mon-go/controllers/controllers.go

```diff
@@ -0,0 +1,244 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/data_timestamp"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/datasets"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/detailed_datasets"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/models"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/mongo"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/responses"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/short_url"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+	"log"
+	"net/http"
+	"time"
+)
+
+var (
+	// ServerInfo defines server info comes from Makefile
+	ServerInfo string
+	// GitVersion git version comes from Makefile
+	GitVersion string
+	Verbose    int
+)
+
+func verboseControllerInitLog(c *gin.Context) {
+	if Verbose > 0 {
+		log.Printf("[INFO] Incoming request to: %s", c.FullPath())
+	}
+}
+
+func controllerInitialize(c *gin.Context) (context.Context, context.CancelFunc, time.Time, models.DataTableCustomRequest) {
+	verboseControllerInitLog(c)
+	start := time.Now()
+	ctx, cancel := context.WithTimeout(context.Background(), mongo.Timeout)
+
+	// Get request json with validation
+	r := models.DataTableCustomRequest{}
+	if err := c.ShouldBindBodyWith(&r, binding.JSON); err != nil {
+		tempReqBody, _ := c.Get(gin.BodyBytesKey)
+		utils.ErrorResponse(c, "Bad request", err, string(tempReqBody.([]byte)))
+	}
+	return ctx, cancel, start, r
+}
+
+func verboseControllerOutLog(start time.Time, name string, req any, data any) {
+	if Verbose > 0 {
+		elapsed := time.Since(start)
+		req, _ := json.Marshal(req)
```

Comment:
please do not skip error checking, especially when parsing json data. It is easy to miss a mailformed json data in this case. I suggest you replace `_` with `err` and check it. According to logic of your code, if it exists you can call `log.Fatal` in that case to provide details about the error.